frontend/src/metabase/static-viz/components/XYChart/utils/ticks.ts

```diff
@@ -67,6 +67,7 @@ export const getXTickWidthLimit = (
 };
 
 export const getXTicksDimensions = (
+  xTickValues: string[] | undefined,
```

Comment:
The gist of passing an extra `xTickValues` is that `xTickValues` is a value that's calculated based on the result of this function, in a sense, this is a cyclic dependency. But in reality `xTickValues` shouldn't have anything to do with dimension right? It's just a part of the existing API that I can't just get the value of `xTickValues` before calling `getXTicksDimensions` the first time easily. So, this is a hack for it.

`getXTicksDimensions` will be called 2 times, and the second time it will get `xTickValues` which will then return the correct margin values calculated based on the values that's displaying on the screen rather than values that might not be displaying on the screen.

See this chart as an example.
![image](https://user-images.githubusercontent.com/1937582/191035524-3a778118-9c81-4348-a463-a9dcba40ae45.png)

Previously, the code will calculate `maxTextWidth` based on all the tick values, rather than what was displayed on the screen. So, it's possible that its value is larger than what it's supposed to be.

e.g. `September 1, 2019` is longer than all the displaying labels here.